### ISSUE TITLE
[Bug]: 在配置文件设置lower_case_table_names模式未生效

### ISSUE THREAD

**Issue by yangj1211** [matrixorigin/matrixone#15031]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main

### Commit ID

newest

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

<img width="605" alt="image" src="https://github.com/matrixorigin/matrixone/assets/153493538/14cfc767-94de-49bc-a209-077429e4ea3c">

在配置文件设置lower_case_table_names为1，命令行查询未生效。

### Steps to Reproduce

```Markdown
1、在cn.toml文件增加配置：
[cn.frontend]
lowerCaseTableNames = 0
2、启动mo
3、连接mo查询lower_case_table_names：
show variables like 'lower_Case_table_names';
```

**Comment by YANGGMM** [comment]
fixed plz test, now can set lowerCaseTableNames use:
```shell
[cn.frontend]
lowerCaseTableNames = "0"
```

**Comment by aronchanisme** [comment]
`sql-mode` works as expected, while `lowerCaseTableNames` doesn't.   
PS: Please note that it's not acceptable if changing confs only take effects when deleting mo-data, as this is completly not a feasible solution in PRD env.

```bash
github@test0:/data/mo/main/matrixone$ mo_ctl connect
2024-03-25 11:12:28.813 UTC+0800    [INFO]    Checking connectivity
2024-03-25 11:12:28.885 UTC+0800    [INFO]    Ok, connecting for user ... 
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 124416
Server version: 8.0.30-MatrixOne-v1.1.1 MatrixOne

Copyright (c) 2000, 2024, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> select git_version();
+---------------+
| git_version() |
+---------------+
| 2dce06dca     |
+---------------+
1 row in set (0.01 sec)

mysql> select @@sql_mode;
+--------------------+
| @@sql_mode         |
+--------------------+
| ONLY_FULL_GROUP_BY |
+--------------------+
1 row in set (0.00 sec)

mysql> show variables like '%lower_case_table_names%';
+------------------------+-------+
| Variable_name          | Value |
+------------------------+-------+
| lower_case_table_names | 1     |
+------------------------+-------+
1 row in set (0.00 sec)

mysql> select @@lower_case_table_names;
+--------------------------+
| @@lower_case_table_names |
+--------------------------+
| 1                        |
+--------------------------+
1 row in set (0.00 sec)

mysql> show variables like '%sql_mode%';
+---------------+--------------------+
| Variable_name | Value              |
+---------------+--------------------+
| sql_mode      | ONLY_FULL_GROUP_BY |
+---------------+--------------------+
1 row in set (0.00 sec)

mysql> create database testDB1;
Query OK, 1 row affected (0.06 sec)

mysql> create database testdb1;
ERROR 1007 (HY000): database testdb1 already exists
mysql> create table testdb1.test(id int, age int);
Query OK, 0 rows affected (0.03 sec)

mysql> insert into testdb1.test values (1, 20), (2, 30), (3, 25);
Query OK, 3 rows affected (0.01 sec)

mysql> select * from testdb1.test;
+------+------+
| id   | age  |
+------+------+
|    1 |   20 |
|    2 |   30 |
|    3 |   25 |
+------+------+
3 rows in set (0.00 sec)

mysql> select id, max(age) from testdb1.test;
ERROR 1149 (HY000): SQL syntax error: column "test.id" must appear in the GROUP BY clause or be used in an aggregate function
mysql> exit
Bye
2024-03-25 11:14:10.510 UTC+0800    [INFO]    Connect succeeded and finished. Bye

github@test0:/data/mo/main/matrixone$ mo_ctl restart 
2024-03-25 11:15:38.960 UTC+0800    [INFO]    At least one mo-service is running. Process info: 
github   1193264       1 33 11:14 ?        00:00:21 /data/mo/main/matrixone/mo-service -daemon -debug-http :9876 -launch /data/mo/main/matrixone/etc/launch/launch.toml
2024-03-25 11:15:38.996 UTC+0800    [INFO]    List of pid(s): 
1193264
2024-03-25 11:15:39.032 UTC+0800    [INFO]    Try stop all mo-services found for a maximum of 10 times, try no: 1
2024-03-25 11:15:39.068 UTC+0800    [INFO]    Stopping mo-service with pid 1193264 with command: kill  1193264
2024-03-25 11:15:39.105 UTC+0800    [INFO]    Wait for 5 seconds
2024-03-25 11:15:44.195 UTC+0800    [INFO]    No mo-service is running
2024-03-25 11:15:44.227 UTC+0800    [INFO]    Stop succeeded
2024-03-25 11:15:44.260 UTC+0800    [INFO]    Wait for 2 seconds
2024-03-25 11:15:46.351 UTC+0800    [INFO]    No mo-service is running
2024-03-25 11:15:46.397 UTC+0800    [DEBUG]    Check total memory on current machine, command: free -m | awk 'NR==2{print }', result(Mi): 15996
2024-03-25 11:15:46.444 UTC+0800    [INFO]    Get conf succeeded: MO_DEPLOY_MODE="git"
2024-03-25 11:15:46.484 UTC+0800    [INFO]    GO memory limit(Mi): 14396
2024-03-25 11:15:46.516 UTC+0800    [DEBUG]    Start command will add GOMEMLIMIT=14396MiB
2024-03-25 11:15:46.548 UTC+0800    [INFO]    Starting mo-service: cd /data/mo/main/matrixone/ && GOMEMLIMIT=14396MiB /data/mo/main/matrixone/mo-service -daemon -debug-http :9876 -launch /data/mo/main/matrixone/etc/launch/launch.toml >/data/mo/main/matrixone/logs/stdout-20240325_111546.log 2>/data/mo/main/matrixone/logs/stderr-20240325_111546.log
2024-03-25 11:15:46.719 UTC+0800    [INFO]    Wait for 2 seconds
2024-03-25 11:15:48.809 UTC+0800    [INFO]    At least one mo-service is running. Process info: 
github   1193755       1 99 11:15 ?        00:00:02 /data/mo/main/matrixone/mo-service -daemon -debug-http :9876 -launch /data/mo/main/matrixone/etc/launch/launch.toml
2024-03-25 11:15:48.843 UTC+0800    [INFO]    List of pid(s): 
1193755
2024-03-25 11:15:48.878 UTC+0800    [INFO]    Start succeeded
github@test0:/data/mo/main/matrixone$ cat /data/mo/main/matrixone/etc/launch/cn.toml 
service-type = "CN"
data-dir = "./mo-data"

[log]
level = "info"
# disable-store, 默认为 false. 即MO会保存所有日志到 system.log_info表
disable-store = true

[cn]
uuid = "dd1dccb4-4d3c-41f8-b482-5251dc7a41bf"
port-base = 18000

[observability]
# disableTrace, 默认为 false。即，会上报 trace、log、error的信息
disableTrace = true

[cn.frontend]
lowerCaseTableNames = "0"
sql-mode = "ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION,NO_ZERO_DATE,NO_ZERO_IN_DATE,STRICT_TRANS_TABLES"
github@test0:/data/mo/main/matrixone$ mo_ctl connect
2024-03-25 11:15:58.594 UTC+0800    [INFO]    Checking connectivity
2024-03-25 11:15:58.688 UTC+0800    [INFO]    Ok, connecting for user ... 
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 124607
Server version: 8.0.30-MatrixOne-v1.1.1 MatrixOne

Copyright (c) 2000, 2024, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> show variables like '%lower_case_table_names%';
+------------------------+-------+
| Variable_name          | Value |
+------------------------+-------+
| lower_case_table_names | 1     |
+------------------------+-------+
1 row in set (0.00 sec)

mysql> show variables like '%sql_mode%';
+---------------+----------------------------------------------------------------------------------------------------+
| Variable_name | Value                                                                                              |
+---------------+----------------------------------------------------------------------------------------------------+
| sql_mode      | ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION,NO_ZERO_DATE,NO_ZERO_IN_DATE,STRICT_TRANS_TABLES |
+---------------+----------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> select * from testdb1.test;
+------+------+
| id   | age  |
+------+------+
|    1 |   20 |
|    2 |   30 |
|    3 |   25 |
+------+------+
3 rows in set (0.01 sec)

mysql> select id, max(age) from testdb1.test;
+------+----------+
| id   | max(age) |
+------+----------+
|    1 |       30 |
+------+----------+
1 row in set (0.00 sec)

mysql> 
```

**Comment by YANGGMM** [comment]
still working

**Comment by YANGGMM** [comment]
fixed

**Comment by aronchanisme** [comment]
Fixed in both `main` and `1.1-dev`

1. `main`-`b7f90bda5`
```bash
aronchen@momac etc % cat ./launch/cn.toml 
service-type = "CN"
data-dir = "./mo-data"

[log]
level = "info"

[cn]
uuid = "dd1dccb4-4d3c-41f8-b482-5251dc7a41bf"
port-base = 18000

[cn.frontend]
lowerCaseTableNames = "0"

aronchen@momac etc % pwd 
/data/mo/main/matrixone/etc
aronchen@momac etc % mo_ctl get_branch
2024-04-12 10:53:38.425 UTC+0800    [INFO]    Try get mo branch
2024-04-12 10:53:38.475 UTC+0800    [INFO]    Get branch succeeded, current branch: main
aronchen@momac etc % mo_ctl get_cid
2024-04-12 10:53:40.286 UTC+0800    [INFO]    Try get mo commit id
commit b7f90bda598cd77f5b0ab880f8b3bea906c3122d
Author: iamlinjunhong <49111204+iamlinjunhong@users.noreply.github.com>
Date:   Fri Apr 12 10:08:53 2024 +0800

    fix validate lock service (#15438)
    
    fix validate lock service
    
    Approved by: @zhangxu19830126
2024-04-12 10:53:40.334 UTC+0800    [INFO]    Get commit id succeeded
aronchen@momac etc % mo_ctl start 
2024-04-12 10:53:43.292 UTC+0800    [INFO]    No mo-service is running
2024-04-12 10:53:43.351 UTC+0800    [INFO]    GO memory limit(Mi): 14745
2024-04-12 10:53:43.408 UTC+0800    [INFO]    Starting mo-service: cd /data/mo/main/matrixone/ && GOMEMLIMIT=14745MiB /data/mo/main/matrixone/mo-service -daemon -debug-http :9876 -launch /data/mo/main/matrixone/etc/launch/launch.toml >/data/mo/main/matrixone/logs/stdout-20240412_105343.log 2>/data/mo/main/matrixone/logs/stderr-20240412_105343.log
2024-04-12 10:53:43.571 UTC+0800    [INFO]    Wait for 2 seconds
2024-04-12 10:53:45.701 UTC+0800    [INFO]    At least one mo-service is running. Process info: 
  501 47524     1   0 10:53AM ??         0:00.11 /data/mo/main/matrixone/mo-service -daemon -debug-http :9876 -launch /data/mo/main/matrixone/etc/launch/launch.toml
2024-04-12 10:53:45.727 UTC+0800    [INFO]    List of pid(s): 
47524
2024-04-12 10:53:45.754 UTC+0800    [INFO]    Start succeeded
aronchen@momac etc % mo_ctl connect
2024-04-12 10:59:52.876 UTC+0800    [INFO]    Checking connectivity
2024-04-12 10:59:52.939 UTC+0800    [INFO]    Ok, connecting for user ... 
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 127
Server version: 8.0.30-MatrixOne-v7905 MatrixOne

Copyright (c) 2000, 2023, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> select git_version();
+---------------+
| git_version() |
+---------------+
| b7f90bda5     |
+---------------+
1 row in set (0.00 sec)

mysql> show variables like "%lower%";
+------------------------+-------+
| Variable_name          | Value |
+------------------------+-------+
| lower_case_file_system | off   |
| lower_case_table_names | 0     |
+------------------------+-------+
2 rows in set (0.01 sec)

mysql> create table Tt (Aa int);
ERROR 1046 (HY000): not connect to a database
mysql> create database if not exists test;
Query OK, 1 row affected (0.01 sec)

mysql> use test;
Database changed
mysql> create table Tt (Aa int);
Query OK, 0 rows affected (0.02 sec)

mysql> insert into Tt values (1), (2), (3);
Query OK, 3 rows affected (0.00 sec)

mysql> select Aa from Tt;
+------+
| aa   |
+------+
|    1 |
|    2 |
|    3 |
+------+
3 rows in set (0.01 sec)

mysql> create table TT (c1 int);
Query OK, 0 rows affected (0.02 sec)

mysql> show tables;
+----------------+
| Tables_in_test |
+----------------+
| TT             |
| Tt             |
+----------------+
2 rows in set (0.00 sec)

mysql> alter table TT add column c2 int;
ERROR 1064 (HY000): SQL parser error: table "tt" does not exist
mysql> alter table `TT` add column c2 int;
ERROR 1064 (HY000): SQL parser error: table "tt" does not exist
mysql> select * from TT;
Empty set (0.00 sec)

mysql> select * from `TT`;
Empty set (0.00 sec)

mysql> select * from `Tt`;
+------+
| Aa   |
+------+
|    1 |
|    2 |
|    3 |
+------+
3 rows in set (0.00 sec)
```

2. `1.1-dev`-`e0d9dad9a`
```bash
aronchen@momac matrixone % cat ./etc/launch/cn.toml 
service-type = "CN"
data-dir = "./mo-data"

[log]
level = "info"

[cn]
uuid = "dd1dccb4-4d3c-41f8-b482-5251dc7a41bf"
port-base = 18000

[cn.frontend]
lowerCaseTableNames = "0"
aronchen@momac matrixone % mo_ctl get_branch
2024-04-12 11:18:06.418 UTC+0800    [INFO]    Try get mo branch
2024-04-12 11:18:06.474 UTC+0800    [INFO]    Get branch succeeded, current branch: 1.1-dev
aronchen@momac matrixone % mo_ctl get_cid 
2024-04-12 11:18:08.490 UTC+0800    [INFO]    Try get mo commit id
commit e0d9dad9a82214ce72f024c5c02aa46fb4fc4ab1
Author: iamlinjunhong <49111204+iamlinjunhong@users.noreply.github.com>
Date:   Thu Apr 11 23:44:45 2024 +0800

    modify defaultMaxLockRowCount to reduce deadlocks (#15472)
    
    modify defaultMaxLockRowCount to reduce deadlocks
    
    Approved by: @zhangxu19830126, @sukki37
2024-04-12 11:18:08.545 UTC+0800    [INFO]    Get commit id succeeded
aronchen@momac matrixone % mo_ctl restart 
2024-04-12 11:18:13.339 UTC+0800    [INFO]    No mo-service is running
2024-04-12 11:18:13.366 UTC+0800    [INFO]    No need to stop mo-service
2024-04-12 11:18:13.394 UTC+0800    [INFO]    Stop succeeded
2024-04-12 11:18:13.421 UTC+0800    [INFO]    Wait for 2 seconds
2024-04-12 11:18:15.569 UTC+0800    [INFO]    No mo-service is running
2024-04-12 11:18:15.635 UTC+0800    [INFO]    GO memory limit(Mi): 14745
2024-04-12 11:18:15.688 UTC+0800    [INFO]    Starting mo-service: cd /data/mo/1.1-dev/matrixone/ && GOMEMLIMIT=14745MiB /data/mo/1.1-dev/matrixone/mo-service -daemon -debug-http :9876 -launch /data/mo/1.1-dev/matrixone/etc/launch/launch.toml >/data/mo/1.1-dev/matrixone/logs/stdout-20240412_111815.log 2>/data/mo/1.1-dev/matrixone/logs/stderr-20240412_111815.log
2024-04-12 11:18:15.855 UTC+0800    [INFO]    Wait for 2 seconds
2024-04-12 11:18:18.003 UTC+0800    [INFO]    At least one mo-service is running. Process info: 
  501 53039     1   0 11:18AM ??         0:00.12 /data/mo/1.1-dev/matrixone/mo-service -daemon -debug-http :9876 -launch /data/mo/1.1-dev/matrixone/etc/launch/launch.toml
2024-04-12 11:18:18.030 UTC+0800    [INFO]    List of pid(s): 
53039
2024-04-12 11:18:18.059 UTC+0800    [INFO]    Start succeeded
aronchen@momac matrixone %  mo_ctl connect
2024-04-12 11:44:13.716 UTC+0800    [INFO]    Checking connectivity
2024-04-12 11:44:13.787 UTC+0800    [INFO]    Ok, connecting for user ... 
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 453
Server version: 8.0.30-MatrixOne-v1.1.2 MatrixOne

Copyright (c) 2000, 2023, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> show variables like "%lower%";
+------------------------+-------+
| Variable_name          | Value |
+------------------------+-------+
| lower_case_file_system | off   |
| lower_case_table_names | 0     |
+------------------------+-------+
2 rows in set (0.00 sec)

mysql> create database if not exists test;
Query OK, 1 row affected (0.01 sec)

mysql> use test;
Database changed
mysql> drop table if exists Tt;
Query OK, 0 rows affected (0.01 sec)

mysql> drop table if exists TT;
Query OK, 0 rows affected (0.00 sec)

mysql> create table Tt (Aa int);
Query OK, 0 rows affected (0.02 sec)

mysql> insert into Tt values (1), (2), (3);
Query OK, 3 rows affected (0.00 sec)

mysql> select Aa from Tt;
+------+
| Aa   |
+------+
|    1 |
|    2 |
|    3 |
+------+
3 rows in set (0.00 sec)

mysql> create table TT (c1 int);
Query OK, 0 rows affected (0.01 sec)

mysql> show tables;
+----------------+
| Tables_in_test |
+----------------+
| Tt             |
| TT             |
+----------------+
2 rows in set (0.00 sec)

mysql> alter table TT add column c2 int; -- should work
ERROR 1064 (HY000): SQL parser error: table "tt" does not exist
mysql> alter table `TT` add column c3 int; -- should work as well
ERROR 1064 (HY000): SQL parser error: table "tt" does not exist
mysql> select * from TT;
Empty set (0.00 sec)

mysql> select * from `TT`;
Empty set (0.00 sec)

mysql> select * from Tt;
+------+
| Aa   |
+------+
|    1 |
|    2 |
|    3 |
+------+
3 rows in set (0.01 sec)

mysql> select * from `Tt`;
+------+
| Aa   |
+------+
|    1 |
|    2 |
|    3 |
+------+
3 rows in set (0.00 sec)

mysql> select git_version();
+---------------+
| git_version() |
+---------------+
| e0d9dad9a     |
+---------------+
1 row in set (0.01 sec)
```
